x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -2287,4 +2296,17 @@ List<ReservedClusterStateHandler<?>> reservedClusterStateHandlers() {
     OperatorPrivileges.OperatorPrivilegesService getOperatorPrivilegesService() {
         return operatorPrivilegesService.get();
     }
+
+    @Override
+    public void close() {
+        if (enabled) {
+            closableComponents.get().forEach(component -> {
```

Comment:
nit: I thin you can use `IOUtils.close`. Don't worry about throwing an exception, I think they're all eventually catched here: https://github.com/elastic/elasticsearch/blob/0ae817e5b60d3801cb02a2b7586573dd8ca72b1d/server/src/main/java/org/elasticsearch/bootstrap/Elasticsearch.java#L468 .

Reply:
Unfortunately, I cannot use `IOUtils.close` because it accepts only `Closeable` objects and our metric implementations are [implementing `AutoCloseable`](https://github.com/elastic/elasticsearch/blob/489aac90e05c49dd3a643639cff4de0b1ae95e77/server/src/main/java/org/elasticsearch/telemetry/metric/LongAsyncCounter.java#L14C35-L14C68).

But I did [change `ApiKeyService`](https://github.com/elastic/elasticsearch/pull/109078/commits/490c8559875d94eb857df6b8062b66eda5d6d33a) to extent `Closeable` since it makes more sense here (plus it removes the IDE warnings about `AutoCloseable` being used without try-with-resources statement ).